script/vsts/platforms/templates/cache.yml

```diff
@@ -9,22 +9,22 @@ parameters:
 
 steps:
   - task: Cache@2
-    displayName: Cache node_modules
+    displayName: Cache apm/node_modules
     inputs:
-      key: 'npm | "$(Agent.OS)" | "$(BUILD_ARCH)" | package.json, package-lock.json, script/vsts/platforms/${{ parameters.OS }}.yml'
-      path: 'node_modules'
-      cacheHitVar: MainNodeModulesRestored
+      key: 'npm_apm | "$(Agent.OS)" | "$(BUILD_ARCH)" | package.json, package-lock.json, apm/package.json, script/package.json, script/package-lock.json, script/vsts/platforms/${{ parameters.OS }}.yml, script/vsts/platforms/templates/preparation.yml'
```

Comment:
Why you have changed this from `npm` to `npm_apm`

Reply:
With identical `key` (looking at the same files and strings), it doesn't work. I tried. There needs to be something different between them.

Otherwise during saving caches, it saves the first one okay :check:, but then the second and third caches say "a cache hit was found" or something, and don't acutally save anything. Then the macOS and Windows test jobs fail, because `script/node_modules` is not restored (totally empty) and the `require('module')` in `script/test` doesn't work.

Reply:
The `npm` thing is just a `string` by the way. Its exact value isn't important, just that it be the same between runs so you can hit the same cache matching the same `key`.

https://docs.microsoft.com/en-us/azure/devops/pipelines/release/caching?view=azure-devops#key-input

Reply:
Here was the (failing) CI run with `npm` as the string at the beginning all three cache `key`s.

https://dev.azure.com/DeeDeeG/b/_build/results?buildId=613&view=results

Notice that all of the test jobs had issues due to improperly saved/not properly restorable caches. That's what having different strings for each cache solves.

Reply:
Last comment about this, because it's four comments in a row now...

I could change it to `npm | apm`, `npm | script` and `npm | main` if you like. A bit more readable that way?